coffee/plan.py

```diff
@@ -176,6 +178,8 @@ def _generate_cpu_code(self, kernel, **kwargs):
         else:
             params = opts
 
+        update_citations(params)
```

Comment:
Mm that's true. I could mimic this logic in Firedrake, but Firedrake only "sees" O2 and O3...and that may become a problem as well....?

OK, in Firedrake I could have (I guess in parameters.py):
O1 -> pick Y
O2 -> pick X
O3 -> pick Y 
if something different than O{1,2,3}, then the user is probably in the business of performance optimisation, so we may (i) just rely on the fact that he will find out what the proper citations are or (ii) issue a warning that recalls to clean the cache to get the citations (clean cache => COFFEE will add the citations through the very same mechanism we have now with this PR)

Other options ? Maybe there's a simpler/better way, but this is what I had on top of my head